# HG changeset patch
# User Jan Beulich <jbeulich@suse.com>
# Date 1365611252 -7200
# Node ID 07b3d86931159d8d92fdf4805876e04cd8f5929a
# Parent  333e2fd4f524706f4921c58cfb1e14ea1edc2a13
x86: show handler for Xen-internal interrupts

... in 'i' debug key output.

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r 333e2fd4f524 -r 07b3d8693115 xen/arch/x86/irq.c
--- a/xen/arch/x86/irq.c
+++ b/xen/arch/x86/irq.c
@@ -17,6 +17,7 @@
 #include <xen/compat.h>
 #include <xen/iocap.h>
 #include <xen/iommu.h>
+#include <xen/symbols.h>
 #include <xen/trace.h>
 #include <xsm/xsm.h>
 #include <asm/msi.h>
@@ -2094,9 +2095,7 @@ static void dump_irqs(unsigned char key)
         if ( ssid )
             printk("Z=%-25s ", ssid);
 
-        if ( !(desc->status & IRQ_GUEST) )
-            printk("mapped, unbound\n");
-        else
+        if ( desc->status & IRQ_GUEST )
         {
             action = (irq_guest_action_t *)desc->action;
 
@@ -2124,6 +2123,10 @@ static void dump_irqs(unsigned char key)
 
             printk("\n");
         }
+        else if ( desc->action )
+            print_symbol("%s\n", (unsigned long)desc->action->handler);
+        else
+            printk("mapped, unbound\n");
 
         spin_unlock_irqrestore(&desc->lock, flags);
 
